### DISCUSSION TITLE
Katana is not working inside shell loop

### DISCUSSION THREAD

**Discussion by bormaa** [projectdiscovery/katana#440] [Q&A]
While i was trying to automate katana with my recon and scanning code 
I saw that after reaching katana command 
`    katana -u $domain -o $domain.txt 
`inside a loop like this
 ```
  while read -r domain; do
    {
    katana -u $domain -o $domain.txt 
       }
    done < domains.txt

```
The shell script stops working suddenly when katana first run with first domain
for example if domains.txt
have 
example.com
example2.com
example3.com 
It will only work for example.com
then script exits

**Comment by ehsandeep** [comment]
@bormaa, can't you use katana directly for this? `katana -list domains.txt`, if you wanted to ensure to process only one domain at a time, you can set the `-p` option to `1`, i.e `katana -list domains.txt -p 1`

**Comment by bormaa** [comment]
I have too loops inside each other one for domains and other for subdomains 
I edited my code to make the call inside the loop of domains 
`   katana -list $pwd/$targetfolder/$url/recon/finalalive.txt -o $pwd/$targetfolder/$url/scan/katana/output.txt
`
The loop exits after finishing the first domain.. The same problem happened before

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
